### PR TITLE
Unit tests: download AMP validator file beforehand

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

This should make running AMP validation JS unit tests more robust by first downloading the AMP Validator file locally in a temp folder so it's readily available for all tests.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Running for example `npm run test:js packages/output/src/test/` multiple times locally should never error or time out.

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: .github/workflows/tests-unit-js.yml [hunk 1]
```diff
@@ -66,10 +65,18 @@ jobs:
         env:
           PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: true
 
+      - name: Download AMP validator JS
+        id: amp_validator
+        run: |
+          cd $RUNNER_TEMP && curl -O -f -s -S https://cdn.ampproject.org/v0/validator_wasm.js
+          AMP_VALIDATOR_FILE="${RUNNER_TEMP}/validator_wasm.js"
+          echo "::set-output name=validator_file::$AMP_VALIDATOR_FILE"
```
Comment (spacedmonkey) [inline .github/workflows/tests-unit-js.yml:73]: This is a 7.2mb file. Anyway we can cache this?

**Comment by spacedmonkey** [comment]
I am still seeing this error. 

```
  ● Text Element output › AMP validation › should produce valid AMP output when setting text letter-spacing

    Expected value   undefined
    Received:
      undefined

    Message:
      Assertion failed: WebAssembly is uninitialized

    Difference:

    Compared values have no visual difference.
        at new module$contents$goog$asserts_
```

In all the amp validation tests. 
Screenshot 2022-06-27 at 15 00 45

**Review by timarney — APPROVED** [review]
Ran multiple times not seeing any issues locally

**Comment by swissspidy** [comment]
@spacedmonkey What's the full command you're using to run tests locally?

If you're using PhpStorm/IntelliJ to run the tests, you're most likely running `node_modules/.bin/jest ...` directly, in which case you will not benefit from the change here (which only works when using `npm run test:js`)

In this case it's up to you to change your test config to either use `npm run test:js` or manually execute the `npm run pretest:js` script in the "Before launch" section, which will then cache the validator script for you.

Like so:

Screenshot 2022-06-27 at 20 10 30

File: .github/workflows/tests-unit-js.yml (hunk 1 above)
Comment (swissspidy) [reply .github/workflows/tests-unit-js.yml:73]: You mean between runs or ..?

I don't see an issue with downloading this anew for every CI run, ensures we're always using the latest version.

We could perhaps cache it with a very short expiration time or something, but that wouldn't really achieve anything.

**Comment by spacedmonkey** [comment]
@swissspidy Still getting the editor locally. 
Screenshot 2022-06-29 at 17 17 23

Screenshot 2022-06-29 at 17 28 01

**Pull request merged**